budget-binder-multiplatform-app/src/commonMain/kotlin/de/hsfl/budgetBinder/presentation/viewmodel/entry/EntryViewModel.kt

```diff
@@ -0,0 +1,226 @@
+package de.hsfl.budgetBinder.presentation.viewmodel.entry
+
+import de.hsfl.budgetBinder.common.Category
+import de.hsfl.budgetBinder.common.DataResponse
+import de.hsfl.budgetBinder.common.Entry
+import de.hsfl.budgetBinder.domain.usecase.*
+import de.hsfl.budgetBinder.presentation.Screen
+import de.hsfl.budgetBinder.presentation.event.UiEvent
+import de.hsfl.budgetBinder.presentation.flow.DataFlow
+import de.hsfl.budgetBinder.presentation.flow.RouterFlow
+import de.hsfl.budgetBinder.presentation.flow.UiEventSharedFlow
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlin.math.absoluteValue
+
+class EntryViewModel(
```

Comment:
Könnte man sich jetzt drüber streiten, eigentlich sollten wir für jede View ein eigenes ViewModel bauen. Also auch für Edit, Create und Co. Schau dir sonst mein CategoryViewModel an. Da habe ich die Anfragen, States und so, was von den anderen Screen verwendet werden kann, in des Haupt `ViewModel` getan, und dies an die unteren vererbt.

Reply:
Ich persönlich mag es mehr das geteilte ViewModel zu haben, da sich edit und create fast gar nicht unterscheiden und es für overview nicht lohnt ein extra ViewModel zu machen. Wäre Entry etwas größer, wäre ich auch für eine Aufteilung in unterViewModel, doch bei 3 Screens und 6 UseCases passt es noch gut in eine Datei. 
Falls du das aber wirklich noch ändern möchtest, dann darfst du das natürlich noch nach den Merge dieser Pull-Request machen, immerhin ist es dann einheitlicher mit den anderen.